### PR TITLE
tell people which MirageOS version to use if they have the wrong one

### PR DESCRIPTION
**Pull request by sabine**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,20 @@
 -include Makefile.config
 
+current_opam_version := $(shell opam --version)
+expected_opam_version:= 2
```
Comment (punchagan) [inline Makefile:4]: I think we need `opam` version >= 2.1 
https://github.com/mirage/mirage/blob/main/mirage.opam#L12

**Comment by sabine** [comment]
Not 100% sure the detailed check for `opam>=2.1.0` works in all compilation environments, please take a careful look, or be prepared to change to a simpler (but not fully accurate) `opam>=2` check.

**Comment by hannesm** [comment]
Thanks for your motivating work. We have since in Mirage 4.4 introduced a header `(* mirage >= 4.4.0 & < 4.6.0 *)` for "config.ml" which outputs a nice error message if your mirage version is too old or too new. Obviously this only works since mirage 4.4.0.

In #395 I added these headers to all the example unikernels :)

**Comment by sabine** [comment]
Thank you, Hannes.

**Pull request closed without merging**